source/applications/advanced/create_depth_map.py

```diff
@@ -1,68 +1,100 @@
 
-    # The Zivid3D.zdf file has to be in the same folder as this sample script.
-    filename_zdf = "Zivid3D.zdf"
+def _point_cloud_to_cv_z(point_cloud):
+    """Get depth map from frame.
 
-    print(f"Reading {filename_zdf} point cloud")
-    frame = zivid.Frame(Path() / f"{str(zivid.environment.data_path())}/{filename_zdf}")
+    Args:
+        point_cloud: Zivid point cloud
 
-    # Getting the point cloud
-    point_cloud = frame.get_point_cloud().to_array()
-    depth_map = np.dstack([point_cloud["z"]])
+    Returns:
+        Depth map (HxWx1 darray)
 
+    """
+    depth_map = point_cloud.copy_data("z")
     depth_map_uint8 = (
-        (depth_map - np.nanmin(depth_map))
-        / (np.nanmax(depth_map) - np.nanmin(depth_map))
-        * 255
+        (depth_map - np.nanmin(depth_map)) / (np.nanmax(depth_map) - np.nanmin(depth_map)) * 255
     ).astype(np.uint8)
 
-    # Applying color map
-    depth_map_color_map = cv2.applyColorMap(depth_map_uint8, cv2.COLORMAP_JET)
+    depth_map_color_map = cv2.applyColorMap(depth_map_uint8, cv2.COLORMAP_VIRIDIS)
 
     # Setting nans to black
-    depth_map_color_map[np.isnan(depth_map)[:, :, 0]] = 0
+    depth_map_color_map[np.isnan(depth_map)[:, :]] = 0
 
-    rgb = np.dstack([point_cloud["b"], point_cloud["g"], point_cloud["r"]])
+    return depth_map_color_map
 
-    # Displaying the RGB image
-    rgb_window = "ImageRGB"
-    cv2.namedWindow(rgb_window, cv2.WINDOW_NORMAL)
-    cv2.imshow(rgb_window, rgb)
 
-    # Waiting for the window to be closed
-    print("Close the RGB image to continue")
-    while cv2.getWindowProperty(rgb_window, 0) >= 0:
-        cv2.waitKey(50)
-    cv2.destroyWindow(rgb_window)
+def _point_cloud_to_cv_bgr(point_cloud):
+    """Get bgr image from frame.
+
+    Args:
+        point_cloud: Zivid point cloud
+
+    Returns:
+        BGR image (HxWx3 darray)
+
+    """
+    rgba = point_cloud.copy_data("rgba")
+    # Applying color map
+    bgr = cv2.cvtColor(rgba, cv2.COLOR_RGBA2BGR)
+
+    return bgr
 
-    # Saving the RGB image
-    cv2.imwrite(f"{rgb_window}.png", rgb)
 
-    # Displaying the Depth map
-    depth_window = "DepthMap"
-    cv2.namedWindow(depth_window, cv2.WINDOW_NORMAL)
-    cv2.imshow(depth_window, depth_map_color_map)
+def _visualize_and_save_image(image, image_file, window_name):
+    """Visualize and save image to file.
 
-    # Waiting for the window to be closed
-    print("Close the Depth map image to continue")
-    while cv2.getWindowProperty(depth_window, 0) >= 0:
+    Args:
+        image: BGR image (HxWx3 darray)
+        image_file: File name
+        window_name: OpenCV Window name
+
+    Returns None
+
+    """
+    cv2.namedWindow(window_name, cv2.WINDOW_NORMAL)
+    cv2.imshow(window_name, image)
+    print("Close the image to continue")
+    while cv2.getWindowProperty(window_name, 0) >= 0:
         cv2.waitKey(50)
-    cv2.destroyWindow(depth_window)
+    cv2.destroyWindow(window_name)
+    cv2.imwrite(image_file, image)
```

Comment:
This doesn't work on my computer. I close the window and nothing happens.
Does it work for anyone else?

Reply:
The "cv2.waitKey(0)" in gamma_correction works, but then the way to move on is to press any key, not close the window.

Reply:
![image](https://user-images.githubusercontent.com/46567886/97873509-08dc2d80-1d18-11eb-8b36-718a757d454e.png)

Reply:
Did you mean that a sample doesn't work or that particular line of code?
![image](https://user-images.githubusercontent.com/46567886/97873790-683a3d80-1d18-11eb-8fac-43b8ccb11a69.png)

![image](https://user-images.githubusercontent.com/46567886/97873844-7720f000-1d18-11eb-8a84-8a6978effd68.png)

Reply:
![image](https://user-images.githubusercontent.com/46567886/97873917-8dc74700-1d18-11eb-9f32-ba591dd145e4.png)

Reply:
What is your suggestion then?

Reply:
I can replace with cv2.waitKey(0) then and appropriate printout

Reply:
Done, check fixup commit. I eddited both gamma and this sample for consistency.

Reply:
Are you saying it works fine on your computer?

Reply:
Ah some github lag here. Lemme check what you did.

Reply:
Both samples work fine now, but only if I "press any key". If I try to hit the "X" on the image window my terminal basically freezes. 

I'd be satisfied if you leave the code as it is now, but change every "Close the image or press any key to continue" to "Press any key to continue".

Reply:
done

Reply:
Nice, thanks!